### PR TITLE
Update lab2 for latest minikube

### LINKED ISSUE #695: [lab2] PVC allocation fails with hostpath provisioner
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

Followed the steps in [lab2](https://kubevirt.io/labs/kubernetes/lab2.html) on a minikube environment, and the `fedora` PVC was stuck in `Pending` state - and so was the importer pod.

**What you expected to happen**:

Lab steps should work flawlessly.

**Anything else we need to know?**:

> If the issue is with a lab, please provide information about your environment, platform, versions, ...

Running this on a Fedora 33 host:

```
$ minikube version
minikube version: v1.17.1
commit: 043bdca07e54ab6e4fc0457e3064048f34133d7e
$ kubectl logs -n kube-system hostpath-provisioner-5f8dd4968f-wccrs | tail -4
I0222 18:42:00.072141       1 leaderelection.go:154] attempting to acquire leader lease...
I0222 18:42:00.098092       1 controller.go:726] claim default/fedora was deleted
E0222 18:42:00.113628       1 leaderelection.go:271] Failed to update lock: Operation cannot be fulfilled on persistentvolumeclaims "fedora": StorageError: invalid object, Code: 4, Key: /registry/persistentvolumeclaims/default/fedora, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: d567e6df-6051-458a-8b65-a7bbaed82310, UID in object meta: 
I0222 18:42:02.817551       1 leaderelection.go:161] stopped trying to acquire lease to provision for pvc default/fedora, task succeeded
```

Note that in minikube there is a default `standard` storage provisioner out of the box, using hostpath.

### PR DESCRIPTION
**Pull request by cwilkers**
- Remove storage-setup.yml and its references
  - Alter pvc_fedora.yml to remove storageclass
  - Download pvc_fedora.yml locally to use it later
  - Add commands to delete and recreate pvc if needed
  - Alter instructions to excise mention of hostpath-provisioner
  - Updated to use minikube ip instead of status

Fixes #695

### CODE REVIEW

**Comment by cwilkers** [comment]
re-pushed to get around spell check not liking "Minikube's"

**Comment by mazzystr** [comment]
@cwilkers You're gonna have to add a commit to fix the following links...
```
- ./_site/2019/prow-jobs-for-kubevirt.html

  *  External link https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/jobs/kubevirt/kubevirt-tutorial/kubevirt-tutorial-presubmits.yaml failed: 404 No error

- ./_site/summit/index.html

  *  External link https://github.com/nunnatsa%E2%80%8B failed: 404 No error
  ```

File: _includes/scriptlets/lab2/06_create_fedora_cloud_instance.sh [hunk 1]
```diff
@@ -1 +1,2 @@
-kubectl create -f https://raw.githubusercontent.com/kubevirt/kubevirt.github.io/master/labs/manifests/pvc_fedora.yml
+wget https://raw.githubusercontent.com/kubevirt/kubevirt.github.io/master/labs/manifests/pvc_fedora.yml
+kubectl create -f pvc_fedora.yml
```
Comment (mazzystr) [inline _includes/scriptlets/lab2/06_create_fedora_cloud_instance.sh:2]: I've never like the idea of making users pulling in yml's

I think we would be better off doing something like ...
```
cat <<EOF | kubectl create -f -
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: "fedora"
  labels:
    app: containerized-data-importer
  annotations:
    cdi.kubevirt.io/storage.import.endpoint: "https://download.fedoraproject.org/pub/fedora/linux/releases/33/Cloud/x86_64/images/Fedora-Cloud-Base-33-1.2.x86_64.raw.xz"
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
  storageClassName: hostpath
EOF
```
That yaml is still doing hostpath.  Is that correct?

File: _includes/scriptlets/lab2/16_minikube_vm_ip.sh [hunk 2]
```diff
@@ -1,5 +1,2 @@
-$ minikube status
-  host: Running
-  kubelet: Running
-  apiserver: Running
-  kubectl: Correctly Configured: pointing to minikube-vm at 192.168.39.74
```
Comment (mazzystr) [inline _includes/scriptlets/lab2/16_minikube_vm_ip.sh:5]: I think we should keep the minikube status to make sure all the pieces of minikube are running

File: labs/kubernetes/lab2.md [hunk 3]
```diff
@@ -16,29 +16,33 @@ You can experiment this lab online at [![Katacoda](/assets/images/katacoda-logo.
```
Comment (mazzystr) [inline labs/kubernetes/lab2.md:16]: There's an English mistake in L15.  

`[CDI](https://github.com/kubevirt/containerized-data-importer) is an utility designed to import Virtual Machine images for use with Kubevirt.`

Would you mind changing that to `is a utility`

File: _includes/scriptlets/lab2/00_delete_and_recreate_pvc.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+kubectl delete -f pvc_fedora.yml --wait
```
Comment (mazzystr) [inline _includes/scriptlets/lab2/00_delete_and_recreate_pvc.sh:1]: Do we really need this for fresh installs?

File: labs/kubernetes/lab2.md [hunk 5]
```diff
@@ -16,29 +16,33 @@ You can experiment this lab online at [![Katacoda](/assets/images/katacoda-logo.
...
 
 ```bash
-{% include scriptlets/lab2/01_get_storage_manifest.sh -%}
-cat storage-setup.yml
-{% include scriptlets/lab2/02_create_storage.sh -%}
 {% include scriptlets/lab2/03_deploy_cdi_operator.sh -%}
 {% include scriptlets/lab2/04_create_cdi-cr.sh -%}
 ```
 
+Check the status of the cdi CustomResource (CR) created in the previous step. The CR's Phase will change from Deploying to Deployed as the pods it deploys are created and reach the Running state.
```
Comment (mazzystr) [inline labs/kubernetes/lab2.md:31]: CDI is a _CustomResourceDefinition_ (CRD)

File: labs/kubernetes/lab2.md [hunk 6]
```diff
@@ -16,29 +16,33 @@ You can experiment this lab online at [![Katacoda](/assets/images/katacoda-logo.
...
+```
+
 Review the "cdi" pods that were added.
 
 ```
 {% include scriptlets/lab2/05_view_cdi_pod_status.sh -%}
 ```
 
-#### Use the CDI
+#### Use CDI to Import a Cloud Image
```
Comment (mazzystr) [inline labs/kubernetes/lab2.md:43]: sed s/Cloud/Disk/

File: labs/kubernetes/lab2.md [hunk 7]
```diff
@@ -52,7 +56,12 @@ This will create the PVC with a proper annotation so that CDI controller detects
 
 Notice that the importer downloaded the publicly available Fedora Cloud qcow image. Once the importer pod completes, this PVC is ready for use in kubevirt.
 
-If the importer pod completes in error, you may need to retry it or specify a different URL to the fedora cloud image. To retry, first delete the importer pod and the {{ site.data.labs_kubernetes_variables.cdi_lab.pvc_name }} PVC, and then recreate the {{ site.data.labs_kubernetes_variables.cdi_lab.pvc_name }} PVC.
+> notes ""
+> If the importer pod completes in error, you may need to retry it or specify a different URL to the fedora cloud image. To retry, first delete the importer pod and the PVC, and then recreate the PVC.
+
+```bash
```
Comment (mazzystr) [inline labs/kubernetes/lab2.md:62]: You should be able to do a code block within the annotation.  Hack on it and let's see how it looks.

**Review by mazzystr — CHANGES_REQUESTED** [review]

File: _includes/scriptlets/lab2/06_create_fedora_cloud_instance.sh (hunk 1 above)
Comment (cwilkers) [reply _includes/scriptlets/lab2/06_create_fedora_cloud_instance.sh:2]: I dropped the storageClassName from the version in the PR.

File: _includes/scriptlets/lab2/16_minikube_vm_ip.sh (hunk 2 above)
Comment (cwilkers) [reply _includes/scriptlets/lab2/16_minikube_vm_ip.sh:5]: Minikube setup is out of scope for this particular lab, there's a separate lab just for that, I thought. The minikube status no longer returns the IP, which is what we're after here.

File: labs/kubernetes/lab2.md (hunk 3 above)
Comment (cwilkers) [reply labs/kubernetes/lab2.md:16]: Sure, (but...) I didn't want to make the PR huge by taking on the job of updating AND editing.

File: _includes/scriptlets/lab2/00_delete_and_recreate_pvc.sh (hunk 4 above)
Comment (cwilkers) [reply _includes/scriptlets/lab2/00_delete_and_recreate_pvc.sh:1]: No, that's part of the Note block for if there's an issue with the link given to CDI (and with the current mirror problem outlined in https://github.com/kubevirt/containerized-data-importer/issues/1737 there is need for knowing this work-around)

File: _includes/scriptlets/lab2/16_minikube_vm_ip.sh (hunk 2 above)
Comment (mazzystr) [reply _includes/scriptlets/lab2/16_minikube_vm_ip.sh:5]: You're right.  Excellent point.

File: labs/kubernetes/lab2.md (hunk 3 above)
Comment (mazzystr) [reply labs/kubernetes/lab2.md:16]: s'ok.  we're editing the next line so might as well pick up a little trash on the way.

**Comment by mazzystr** [comment]
Very good!  Looking better yet!
/approve
/lgtm

**Pull request merged**